Tests/DatadogTests/Datadog/Core/System/NetworkConnectionInfoProviderTests.swift

```diff
@@ -9,51 +9,86 @@ import Network
 @testable import Datadog
 
 class NetworkConnectionInfoProviderTests: XCTestCase {
+    @available(iOS 12, *)
     func testItStartsAndCancelsNWPathMonitor() {
-        let startExpectation = self.expectation(description: "call start")
-        let cancelExpectation = self.expectation(description: "call cancel")
+        let isStarted: (NWPathMonitor) -> Bool = { monitor in
+            return monitor.queue != nil
+        }
+
+        var pathMonitor: NWPathMonitor? = NWPathMonitor()
+        weak var weakMonitor: NWPathMonitor? = pathMonitor
+
+        XCTAssertFalse(isStarted(pathMonitor!))
+
+        var provider: NetworkConnectionInfoProvider? = NetworkConnectionInfoProvider(pathMonitor!)
 
-        let monitor = NWCurrentPathMonitorMock(pathInfo: .mockAny())
-        monitor.onStart = { startExpectation.fulfill() }
-        monitor.onCancel = { cancelExpectation.fulfill() }
+        XCTAssertTrue(isStarted(pathMonitor!))
 
         autoreleasepool {
-            _ = NetworkConnectionInfoProvider(monitor: monitor) // `start()` when initialized
-            wait(for: [startExpectation], timeout: 1)
-        } // `cancel()` when deinitialized
+            pathMonitor = nil
+        }
+        XCTAssertNotNil(weakMonitor, "provider should retain pathMonitor")
+        XCTAssertNotNil(provider!.current, "provider should be working")
+
+        autoreleasepool {
+            provider = nil // `cancel()` when deinitialized
+        }
 
-        wait(for: [cancelExpectation], timeout: 1)
+        XCTAssertNil(provider)
+        XCTAssertNil(weakMonitor, "path monitor should be cancelled and released")
     }
```

Comment:
We no longer test if
```swift
// NetworkConnectionInfoProvider.swift
private let cancelBlock: (() -> Void)?
```
is invoked during deinitialization. How about adding separate initializer capturing `fetchBlock` and `cancelBlock` as parameters and asserting in separate test that `cancelBlock` is actually called during `provider = nil`?

Reply:
IMO `start/cancel` is an implementation detail of `NetworkConnectionInfoProvider`

what we expect from it is simply **keeping the actual monitor alive as long as we need it**
so long as `weakMonitor` is `nil` after `provider = nil`, we shouldn't worry about if `cancel` is called

by the way, if `cancel` is not called, `weakMonitor` doesn't get `nil` and keeps monitoring (guess: `dispatchQueue` may retain it)

Reply:
> by the way, if cancel is not called, weakMonitor doesn't get nil and keeps monitoring (guess: dispatchQueue may retain it)

Ah, I see 👌 - with this it's good to go then 👍.